starboard/shared/win32/posix_emu/socket.cc

```diff
@@ -0,0 +1,92 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <io.h>
+#include <time.h>
+#include <windows.h>
+#include <winsock2.h>
+#include <map>
+
+#include "starboard/common/log.h"
+#include "starboard/types.h"
+
+int gen_fd() {
+  return rand() % (0x7FFFFFFF) + 1;
+}
+
+struct CriticalSection {
+  CriticalSection() { InitializeCriticalSection(&critical_section_); }
+  CRITICAL_SECTION critical_section_;
+};
+
+static std::map<int, SOCKET>* m_addr = nullptr;
+static CriticalSection s_critical_section;
+
+int handle_db_put(SOCKET socket_handle) {
+  EnterCriticalSection(&s_critical_section.critical_section_);
+  if (m_addr == nullptr) {
+    m_addr = new std::map<int, SOCKET>();
+  }
+
+  int fd = gen_fd();
+  while (m_addr->find(fd) != m_addr->end()) {
```

Comment:
Why don't we just keep a counter. This random number generation and checking to find it is used can get expensive.